### DISCUSSION TITLE
Support to ESP32-S3-WROOM-1

### DISCUSSION THREAD

**Discussion by RedSalsicha** [BitMaker-hub/NerdMiner_v2#294] [Q&A] [answered]
I have an ESP32-S3-WROOM-1 (N16R8) here, the same model as the link below.

https://pt.aliexpress.com/item/1005006036815289.html

My question is, can I flash the NerdMiner firmware on it?

But in the flash I choose "ESP32-WROOM" or something else that is S3 like "M5-StampS3" or "LILYGO S3 Dongle"?

Thanks.

**Comment by jsy-rg1570** [comment, marked as answer]
Dear,

You can build "ESP32-S3-devKitv1" it was meant for that the ESP32 S3 Devkit V1 and your board is an clone from that model. 😉

It will work on your board but the RGB led seem to be on the pin 47 whereas the AI-S3 has the pin 48. You can change it in "platformio.ini" before building.

Enjoy

**Reply by RedSalsicha** [reply]
Perfect, after researching how to compile in VS and Platformio, I was able to compile and send the board, and everything worked 100%, thank you.

**Reply by RedSalsicha** [reply]
Was the RGB LED on the ESP32-S3 board meant to blink or something?

Because I saw that on this S3 it doesn't blink like the blue LED on the previous ESP32-WROOM board, I even did a pin test and pin 47 is actually correct, however I edited it to change it to port 47, but it doesn't blink or light up.

Then I had this other related question.

**Reply by jsy-rg1570** [reply]
> Was the RGB LED on the ESP32-S3 board meant to blink or something?
> 
> 
> 
> Because I saw that on this S3 it doesn't blink like the blue LED on the previous ESP32-WROOM board, I even did a pin test and pin 47 is actually correct, however I edited it to change it to port 47, but it doesn't blink or light up.
> 
> 
> 
> Then I had this other related question.

LEDs are off at startup but if you triple click the boot button it should switch on.

**Reply by RedSalsicha** [reply]
Perfect, worked 100%, thank you.

**Reply by sany3001** [reply]
Can you share binary for your device? And maybe advice how to FLASH? I have ESP32-S3-DevKitC-1U and would like to let it mine too.
Or maybe advice only how to start with own build (even better option)?

**Reply by RedSalsicha** [reply]
I would even send the BIN without any problems, but from what I understand, VSCode compiles and sends the files directly to the device, it does not generate the files to be sent via the website.

But you will need a computer with the "Visual Studio Code" program and on it you install the "PlatformIO IDE" extension, with it installed, you download the NerdMiner repository, remove it from the compressed folder (Winrar/Winzip), and at PIO HOME of the "PlatformIO IDE" extension, go to "Import Arduino Project", select the folder you unzipped.

In the program footer you will have an "Env" selected (normally the default "NerdMiner_V2", click there and change it to "ESP32-S3-devKitv1"

Then, with the board plugged into the computer, just go to the "Build" option and then to "Upload", wait for it to finish and that's it, the board will restart with the firmware configured, as it won't have a screen with QRCode, just search for the WIFI and connect to the board and configure it and then follow it on the pool website.

The video below helped me a lot about the "PlatformIO IDE" extension, opening the project and compiling, as well as showing where the options are.. =D

https://www.youtube.com/watch?v=KZZXaDGCWpo

**Reply by sany3001** [reply]
Thanks for the great answer! I have done steps you wrote (only used `git --clone https://github.com/BitMaker-hub/NerdMiner_v2.git` in Linux Mint to get project locally). I will update this answer in any case - either it will be all ok or I will share problems I was dealing with.

Compiled, flashed (did hold boot button on power up, connected to UART microusb on board). I can see NerdMineraAP but I cannot connect to it. I have tried to wipe flash but it did not help. I will keep trying.

**Reply by RedSalsicha** [reply]
Perfect, let me know if you have any questions that might help.
The good thing about doing the Build yourself is that you can edit some things, and with that I pranked my brother by showing the NerdMiner in LILYGO T-Display with the "Valid Block" at 1 so he thought we had won, 😄😄😄

**Reply by sany3001** [reply]
Good joke! Now I know what means "prank". :-)

I have this ouptut from serial line of my miner, not saying much though:
![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/31476550/b556b2a7-a7cc-49de-bfb9-20f2f5311af3)
My miner is without antenna, maybe I will try to connect one.

**Reply by sany3001** [reply]
So seems to be that missing antenna was the real problem. It also corresponds to signal strength was much more, AP was disappearing, now I had 100%.
Just reminding that the kit name is `ESP32-S3-DevKitC-1U` and in PlatformIO I have chosen `ESP32-S3-devKit1`.

PS: I have "hard coded" my BTC address into source code, this was always pain to not forget to copy that.
I can also try to fix issue when my miners activated AP mode even connected to STA and mining.
PPS: I have referenced you in https://github.com/BitMaker-hub/NerdMiner_v2/issues/298

**Reply by OONDY** [reply]
I am having trouble to get Nerdminer to run on the same board as above. 
Do you have a bin file that you used that I could try on my board?

Thanks
Andrew

**Reply by rmarquina1** [reply]
im having the same issue, any ideas?

**Reply by rmarquina1** [reply]
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x2a (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x44c
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a80
entry 0x403c98d0
E (2155) SPIFFS: mount failed, -10025
[  2350][E][SPIFFS.cpp:89] begin(): Mounting SPIFFS failed! Error: -1
E (2156) SPIFFS: mount failed, -10025
[  2352][E][esp32-hal-misc.c:128] disableCore0WDT(): Failed to remove Core 0 IDLE task from WDT

**Comment by Fudskie** [comment]
https://flasher.bitronics.store/ i used this flasher
